inc/Classes/Translation.php

```diff
@@ -32,28 +39,19 @@ class Translation
     ];
 
     /**
-     * Valid languages
+     * Valid languages.
      *
-     * @var array
+     * @var array<string>
      */
-    public $valid_lang = ['de', 'en', 'es', 'fr', 'nl', 'it'];
+    public array $valid_lang = ['de', 'en', 'es', 'fr', 'nl', 'it'];
```

Comment:
`$valid_lang = array_keys($this->lang_names);` ?

Reply:
We could do this. However, I would prefer to get rid of `$this->lang_names` in the future at all.
Main reason: The language names are in german. If I aim to switch the language, the german language names doesn't help me much.
I would like to extract the language names into translation files in the future.

In this PR, I am to rework the class without big functionality changes, but rather make it type safe and limit the public interface.
In the next step, I would prefer to make such optimizations.